ILSpy/Metadata/Helpers.cs

```diff
@@ -212,11 +212,21 @@ static void ApplyAttributes(PropertyDescriptor descriptor, Binding binding, Data
 			}
 		}
 
+		[Obsolete("Use ReadOnlySpan<byte> overload")]
 		[MethodImpl(MethodImplOptions.AggressiveInlining)]
 		public static unsafe int GetValue(byte* ptr, int size)
+			=> GetValue(new ReadOnlySpan<byte>(ptr, size));
+
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		public static int GetValue(ReadOnlySpan<byte> ptr, int size)
+			=> GetValue(ptr.Slice(0, size));
+
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		public static int GetValue(ReadOnlySpan<byte> ptr)
 		{
+			// endianess?
```

Comment:
If I am not mistaken, the code below assumes little endian byte-order and ECMA335 states that big endian byte-order is only used in signatures. Maybe we should call this method "GetValueLittleEndian"?

Reply:
When decoding signatures, we should not need this method, as we already have access to the raw bytes and SRM `BlobReader` already exposes helpers that handle compressed integers.

Reply:
1. renamed `GetValueLittleEndian`
2. unless I'm misunderstanding these are not "compressed", they are just either "small" (2 byte) or "large" (4 bytes) depending on the size of the table they're indexing. Maybe this could be a `ReadTableIndex` or something, but I'm not sure how that would work.

Reply:
I don't follow...

Reply:
I'm sorry, what is unclear? Which method on BlobReader could we use instead of GetValueLittleEndian?

Reply:
OK... now I understand... sorry.... apparently my second comment yesterday was the confusing one: It was just meant as additional information regarding LE vs BE encoding used in ECMA335. BE is only used in signature blobs, in which case we don't need GetValue (now called "GetValueLittleEndian"), because SRMs BlobReader already supports all the different primitive types used in signatures.

I did not mean to suggest using BlobReader instead of GetValue/GetValueLittleEndian. However, now that I think more about it, maybe we could/should use the methods found in BinaryPrimitives instead? Not sure whether that's a good idea for all the different uses of the method.

Reply:
You mean reimplementing `GetValueLittleEndian` in terms of `BinaryPrimitives`, or in each of the callers?

Also, it is public - do we have to go though a obsoletion/deprecation process or can it just be changed/removedv?

Reply:
I don't expect anyone to actually use the method outside of the decompiler engine or ILSpy, however, wouldn't do any harm to deprecate the original overload (the one that uses pointers) and keep the name "GetValue" so we don't break anyone, right?

I think all the uses of GetValue(LittleEndian) could be directly replaced by BinaryPrimitives?